evalml/tests/component_tests/test_oversampler.py

```diff
@@ -504,3 +526,26 @@ def test_oversampler_nullable_type_incompatibility(
         X, y = evalml_oversampler._handle_nullable_types(X, y)
 
     im_oversampler.fit_resample(X, y)
+
+
+@patch(
+    "evalml.pipelines.components.component_base.ComponentBase._handle_nullable_types",
+)
+def test_oversampler_calls_handle_nullable_types(
+    mock_handle_nullable_types,
+    X_y_binary,
+):
+    X, y = X_y_binary
+
+    oversampler = Oversampler()
+
+    assert not mock_handle_nullable_types.called
+    mock_handle_nullable_types.return_value = X, y
+
+    # Incompatibility is only when the sampler is actually called in transform,
+    # so we don't need to handle nullable types at fit
+    oversampler.fit(X, y)
+    assert not mock_handle_nullable_types.called
```

Comment:
Want to highlight that for many of the components with incompatibilities, it's only in one of the fit/predict/transform methods, so I've chosen to not call `_handle_nullable_types` when it's not needed, which matches behavior I've seen used in the past of only making incompatibility-related changes in the exact context they're needed. 

Let me know if anyone has a problem with this, though. I made sure that none of the components have any logic that would change how the component is fit based on the downcasting of types to handle incompatibilities, but I may be missing something.